src/components/template/PdfTemplateEditor.tsx

```diff
@@ -0,0 +1,92 @@
+import { TextField, Typography } from '@mui/material';
+import Button from '@mui/material/Button';
+import { Field, Form, Formik } from 'formik';
+import React, { useEffect, useState } from 'react';
+
+import UOLoader from 'components/common/UOLoader';
+import { PdfTemplate, Template } from 'generated/sdk';
+import {
+  StyledButtonContainer,
+  StyledContainer,
+  StyledPaper,
+} from 'styles/StyledComponents';
+import useDataApiWithFeedback from 'utils/useDataApiWithFeedback';
+
+export default function PdfTemplateEditor() {
+  const { api } = useDataApiWithFeedback();
+  const [loading, setLoading] = useState<boolean>(true);
+  const [template, setTemplate] = useState<Template | null>(null);
+  const [pdfTemplate, setPdfTemplate] = useState<PdfTemplate | null>(null);
+
+  // TODO: get the template ID a better way
+  const templateId = Number(window.location.pathname.split('/').pop());
```

Comment:
Could this be passed as a React prop somewhere?

Reply:
To get it from URL I think this would work
```
import { useParams } from 'react-router';
const { pdfTemplateId } = useParams<{ pdfTemplateId: string }>();
```

Reply:
Great to know, thanks!